lib/rouge/lexers/q.rb

```diff
@@ -65,7 +65,7 @@ def self.builtins
         end
 
         # White space and comments
-        rule(%r{[ \t\r]\/.*$}, Comment::Single)
+        rule(%r{[ \t\r]+\/.*$}, Comment::Single)
         rule(/[ \t\r]+/, Text::Whitespace)
```

Comment:
Is there a reason to include whitespace at all at the beginning of this pattern? Do comments in Q have to begin with whitespace? Can you begin a line with `/ This is a comment`?

Reply:
You can start a line with a comment, but as `/` is a valid operator there needs to be a whitespace between the last character and the `/` for the comment to be recognised,

```q
/ this is my sum function
f:{x+y}
f/1 2 3 4 / this will return 10
f / this will return the function def
```

should look like this:
![image](https://user-images.githubusercontent.com/22226349/63171040-9f003700-c032-11e9-990a-3a9a2145051c.png)

Perhaps that line should be:
```
rule(%r{[ \t\r\n]+\/.*$}, Comment::Single)
```

Reply:
The regexp can be more like the following, IMO:
```
%r(/\s+.*)
```

Reply:
Apologies, I should have checked the demo. The lexer already lexed lines beginning with `/` because they matched the multiline comment rule a bit later.

That's not ideal because they're being tokenised incorrectly but the regular expression I'd want to use (which would do a negative lookbehind) won't work because of limitations in Ruby's StringScanner class.

I did take the opportunity to rebase against the current version of master and to simplify the regex a little. Everything still looked good to me. Does it look OK to you, @streetster?

Reply:
just pulled in your change and took a look at the samples/q - everything looks good - nice work on the simplification!